packages/amplify_auth_cognito/example/lib/Widgets/ViewUserAttributes.dart

```diff
@@ -0,0 +1,119 @@
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:amplify_flutter/amplify.dart';
+import 'package:flutter/material.dart';
+
+import 'UpdateUserAttribute.dart';
+
+// ignore: public_member_api_docs
+class ViewUserAttributes extends StatefulWidget {
+  @override
+  _ViewUserAttributesState createState() => _ViewUserAttributesState();
+}
+
+class _ViewUserAttributesState extends State<ViewUserAttributes> {
+  bool _isLoading = true;
+  String _errorMessage = '';
+  List<AuthUserAttribute> _userAttributes = [];
+
+  void _showSuccess(String message) {
+    ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(backgroundColor: Colors.green[800], content: Text(message)));
+  }
+
+  void _showError(String message) {
+    ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(backgroundColor: Colors.red[900], content: Text(message)));
+  }
+
+  Future _fetchAttributes({bool isRefresh = false}) {
+    return Amplify.Auth.fetchUserAttributes().then((attributes) {
+      setState(() => _userAttributes = attributes
+        ..sort((a, b) => b.userAttributeKey.compareTo(a.userAttributeKey)));
+      if (isRefresh) {
+        _showSuccess('User Attributes Refreshed Successfully');
+      }
+      return;
+    }).catchError((error) {
+      var errorMessage =
+          'There was an error fetching the user attribute: $error';
+
+      if (isRefresh) {
+        _showError(errorMessage);
+      } else {
+        setState(() => _errorMessage = errorMessage);
+      }
+    }).whenComplete(() => setState(() => _isLoading = false));
```

Comment:
I'm not quite sure this is working as intended.  When attempting to view attributes with a user w/ expired session, the loading widget hangs.

Reply:
In Cognito you can view the app client, and then set the refresh token to expire in 60 minutes (and all other tokens in 5).  So... make this change in Cognito, signIn, and wait an hour.